File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/source/reddit/RedditApi.java
package de.jbamberger.offlineclient.source.reddit;

import de.jbamberger.offlineclient.db.entity.RedditPost;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public interface RedditApi {

    String BASE_URL = "https://reddit.com/";


    @GET("r/{subReddit}/")
    Call<RedditPost> getPosts(@Path("subReddit") String subReddit);

    @GET("r/{subReddit}/comments/{postId}/{postName}")
    Call<RedditPost.Comment> getPostComments(@Path("subReddit") String subReddit, @Path("postId") String postId, @Query("postName") String postName);
}

File: OfflineClient/cached-api/src/main/java/de/jbamberger/api/provider/reddit/RedditApi.kt
package de.jbamberger.api.provider.reddit

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

internal interface RedditApi {


    @GET("r/{subReddit}/")
    fun getPosts(@Path("subReddit") subReddit: String): Call<RedditPost>

    @GET("r/{subReddit}/comments/{postId}/{postName}")
    fun getPostComments(@Path("subReddit") subReddit: String, @Path("postId") postId: String, @Query("postName") postName: String): Call<RedditPost.Comment>

    companion object {
        val BASE_URL = "https://reddit.com/"
    }
}